webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultGroupService.java

```diff
@@ -48,7 +56,11 @@ public GroupFilterOptions getFilterOptions(final User user) {
     }
 
     @Override
+    @Transactional
     public void delete(final User user, final long id) {
-        repository.delete(user.getPermissionsById().get("GROUP_WRITE").getScope(), id);
+        final String digest = HexUtils.toHexString(DigestUtils.md5(user.getEmail() + id));
+        final long importId = importRepository.create(digest, "group delete");
+        repository.delete(user.getPermissionsById().get("GROUP_WRITE").getScope(), id, importId);
+        importRepository.trigger(importId);
```

Comment:
This is the big change.  By marking the service method as transactional I don't need to handle fallback cases where the group doesn't exist or the user doesn't have access.